src/components/button-link/index.tsx

```diff
@@ -3,16 +3,22 @@ import s from './button-link.module.css'
 import { ButtonProps } from 'components/button/types'
 import classNames from 'classnames'
 
+type NativeLinkProps = JSX.IntrinsicElements['a']
+
 /**
  * _Note WIP Component_
  * this button link component should mimic the design system options
  * outlined in `Button` component. This is a WIP implementation and should be
  * expanded upon. It currently renders a theme colors and sizes, with styles
- * copied from `Button`. Still needs to support: icons
+ * copied from `Button`.
  **/
 interface ButtonLinkProps
-  extends Pick<ButtonProps, 'color' | 'size' | 'text' | 'ariaLabel'> {
+  extends Pick<
+    ButtonProps,
+    'color' | 'size' | 'text' | 'ariaLabel' | 'icon' | 'iconPosition'
+  > {
   href: string
+  target?: NativeLinkProps['target']
```

Comment:
💭 Should we do [what `StandaloneLink` does](https://github.com/hashicorp/dev-portal/blob/main/src/components/standalone-link/index.tsx) and have an `openInNewTab` prop that handles setting both `target` and `rel` appropriately on the rendered `<a>`?

Reply:
✅  Great call! Done, I took the same approach as `StandaloneLink` 💯 Thanks for the quick reference ✨